### PR TITLE
Create and JIT coerce_II op

### PR DESCRIPTION
**Pull request by MasterDuke17**
Implement it with the new `MVM_bigint_from_bigint` function. This will be
used in Rakudo's `Int.new` to create one Int from another, but without any
mix-ins the source might have.

### CODE REVIEW

File: src/math/bigintops.c [hunk 1]
```diff
@@ -829,6 +829,26 @@ void MVM_bigint_from_str(MVMThreadContext *tc, MVMObject *a, const char *buf) {
     }
 }
 
+MVMObject * MVM_bigint_from_bigint(MVMThreadContext *tc, MVMObject *result_type, MVMObject *a) {
+    MVMObject * const result = MVM_repr_alloc_init(tc, result_type);
```
Comment (jnthn) [inline src/math/bigintops.c]: Need to `MVMROOT` `a` while doing this allocation.

**Comment by MasterDuke17** [comment]
Added the MVMROOT.

File: src/math/bigintops.c [hunk 2]
```diff
@@ -817,6 +817,32 @@ void MVM_bigint_from_str(MVMThreadContext *tc, MVMObject *a, const char *buf) {
     }
 }
 
+MVMObject * MVM_bigint_from_bigint(MVMThreadContext *tc, MVMObject *result_type, MVMObject *a) {
+    MVMP6bigintBody *a_body  = get_bigint_body(tc, a);
```
Comment (jnthn) [inline src/math/bigintops.c]: This returns an interior pointer into `a`, so this assignment should be moved until after the `MVMROOT` also.

Comment (MasterDuke17) [reply src/math/bigintops.c]: Fixed

**Pull request merged**